### PR TITLE
svt-hevc 1.4.3 (new formula)

### PR DESCRIPTION
**Pull request by yurikoles**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by yurikoles** [comment]
Previous discussion: #51724

**Comment by yurikoles** [comment]
Sometimes test may fail due to https://github.com/OpenVisualCloud/SVT-HEVC/issues/495

File: Formula/svt-hevc.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  depends_on "cmake" => :build
+  depends_on "yasm" => :build
+
+  resource "akiyo_qcif.y4m" do
+    url "https://media.xiph.org/video/derf/y4m/akiyo_qcif.y4m"
+    sha256 "df88d83cbf6d99f3ec41f2c1fd2e67665d2a71ff8caa08f8b6bc46bf4567ea2e"
+  end
+
+  def install
+    resource("akiyo_qcif.y4m").stage(pkgshare.to_s)
```
Comment (bayandin) [inline Formula/svt-hevc.rb]: If we need this file only for testing purposes, there is no reason to install it.
You can use `resource("akiyo_qcif.y4m").stage(testpath)` in test block

Comment (yurikoles) [reply Formula/svt-hevc.rb]: Thanks for suggestion!

**Comment by yurikoles** [comment]
This formula requires rising macOS handlers limits. Is this acceptable in brew? Is there a special caveat keyword for this? Could someone suggest a good how-to on this topic, because Intel references a bit outdated doc. I promise to open a PR upstream to update reference on macOS limits.

**Comment by bayandin** [comment]
Homebrews caveats are for homebrew specific things (https://docs.brew.sh/Formula-Cookbook#caveats) if you're talking about something like a maximum number of open file descriptors — this even not macOS specific and shouldn't be handled properly in an application itself (raising proper exception / printing suggestion).

**Comment by yurikoles** [comment]
I'm unsure how to fix tests, because raising limits requires `sudo` and reboot.

**Comment by bayandin** [comment]
Can we use a test which upstream uses on their CI? Does it require to tune limits as well?
[`SvtHevcEncApp -encMode 9 -i akiyo_cif.y4m -b test1.h265 -asm 0`](https://github.com/OpenVisualCloud/SVT-HEVC/blob/d0004f158cf7ccbe5a19c769f2a59e187788e41c/.travis.yml#L63)

Regarding caveats block, since it's not a homebrew specific, then it's not needed.

**Comment by yurikoles** [comment]
@bayandin good suggestion, but actually this seems to do the trick:
[`  - if [ $TRAVIS_OS_NAME = osx ]; then ulimit -n 512; fi`](https://github.com/OpenVisualCloud/SVT-HEVC/blob/d0004f158cf7ccbe5a19c769f2a59e187788e41c/.travis.yml#L55)

**Comment by bayandin** [comment]
We could set `ulimit -n 512` for testing purposes, see an example in [`duplicity`](https://github.com/Homebrew/homebrew-core/blob/a8223c16e949c0c1700c46750203e3e581edc0e9/Formula/duplicity.rb#L281).

**Comment by yurikoles** [comment]
Strange, tests work fine locally on Catalina and ArchLinux.

**Comment by bayandin** [comment]
The test fails with [`Illegal instruction: 4`](https://github.com/Homebrew/homebrew-core/pull/52271/checks?check_run_id=566850213#step:4:703).

I'm not sure, but it could be because we have too old CPU for SVT-HEVC.
- We have `CPU: hexa-core 64-bit penryn`
- [Hardware requirements](https://github.com/OpenVisualCloud/SVT-HEVC#hardware) state `the encoder runs on any 5th Generation Intel Core™ Processors (formerly Broadwell) CPUs (Xeon E5-v4) or newer.`

**Pull request closed without merging**